mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/Client.h

```diff
@@ -131,6 +147,17 @@ class CompilerClient {
   static void setupPassManagerLogging(mlir::PassManager &pm,
                                       const DebugOptions &options);
 
+  StatusOr<std::unique_ptr<mlir::OptionsContext>>
+  createOptions(const std::string &optionsType,
```

Comment:
- The LLVM/MLIR conventions that we follow encourage using `llvm::StringRef` (by value) in place of `const std::string &`.
- Similarly, please use `llvm::ArrayRef<T>` (by value) in place of `const std::vector<T>&`
- Please move implementation to header file.